### ISSUE TITLE
Repeat doesn't interact correctly with autocomplete

### ISSUE THREAD

**Issue by podgib** [atom/vim-mode#650]
When repeating a previous command using '.', the repeated command doesn't include characters added by autocomplete.

Eg.
```C
  int x = f(input);
  int y = f(input);
```

If I move my cursor to the start of 'input', then press cw, 'input' is removed and I am in insert mode. I then start typing 'output', but after 'out', I use autocomplete to finish the word. If I now leave insert mode, move to the next 'input' and press '.', it only inserts 'out', not 'output'.

Expected:
```C
  int x = f(output);
  int y = f(output);
```

Actual:
```C
  int x = f(output);
  int y = f(out);
```

**Comment by jacekkopecky** [comment]
Yes, and it also doesn't play nice with automatic matching of parentheses:
`i(abc)<esc>` can replay as "()abc". I haven't yet had enough time to try
and solve this one. I expect one fix can cover both problems, though.

**Comment by maxbrunsfeld** [comment]
I think this was fixed by #712.
